bookkeeper-server/bin/bookkeeper

```diff
@@ -71,18 +71,18 @@ else
 fi
 
 # exclude tests jar
-RELEASE_JAR=$(ls ${BK_HOME}/*bookkeeper-server-*.jar 2> /dev/null | grep -v tests | tail -1)
+RELEASE_JAR=$(ls ${BK_HOME}/*bookkeeper-server-*.jar 2> /dev/null | grep -v tests | grep -v sources | tail -1)
```

Comment:
if the problem comes from multiple package, or your shell doesn't emit the right result. You need to exclude following packages:

```
target/bookkeeper-server-4.6.0-SNAPSHOT-javadoc.jar
target/bookkeeper-server-4.6.0-SNAPSHOT-shaded.jar
target/bookkeeper-server-4.6.0-SNAPSHOT-sources.jar
```

However I would suggest digging into this issue a bit more and come up a command that can produce determistic results.

Reply:
Strange. Yes, we should make this more deterministic. 

My shell(debian stable, bash) collates differently:
```~/src/bookkeeper $ ls bookkeeper-server/target/
apidocs                                            bookkeeper-server-4.7.0-SNAPSHOT-sources.jar  classes                 maven-archiver
bookkeeper-server-4.7.0-SNAPSHOT.jar               bookkeeper-server-4.7.0-SNAPSHOT-tests.jar    generated-sources       maven-shared-archive-resources
bookkeeper-server-4.7.0-SNAPSHOT-javadoc.jar       checkstyle-checker.xml                        generated-test-sources  maven-status
bookkeeper-server-4.7.0-SNAPSHOT-shaded.jar        checkstyle-result.xml                         invoker                 test-classes
bookkeeper-server-4.7.0-SNAPSHOT-shaded-tests.jar  checkstyle-suppressions.xml                   javadoc-bundle-options
```

Reply:
Maybe this is why I was seeing the shading problem also.

Reply:
the result returned by `ls` is unsorted. that means it will have different behaviors on different bash (mac vs linux).

a deterministic way is to `ls ... | sort | tail -1`

Reply:
I assumed ls would lexically sort by default, but I guess that would make ls very expensive on large directories. In any case, I think I'll either check that there's a version before the ".jar" or maybe directly use the shaded. This may fix #773 also.